core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -275,7 +293,23 @@ public HttpFeed() {
     protected HttpFeed(Builder builder) {
         setConfig(ONLY_IF_SERVICE_UP, builder.onlyIfServiceUp);
         Map<String,String> baseHeaders = ImmutableMap.copyOf(checkNotNull(builder.headers, "headers"));
-        
+
+        HttpExecutor httpExecutor = null;
+        if (builder.httpExecutor != null) {
+            httpExecutor = builder.httpExecutor;
+        } else {
+            HttpExecutorFactory httpExecutorFactory = null;
+            Collection<Location> locations = builder.entity.getLocations();
+            Maybe<MachineLocation> location =  Machines.findUniqueMachineLocation(locations, MachineLocation.class);
+            if (location.isPresent() && location.get().hasExtension(HttpExecutorFactory.class)) {
+                httpExecutorFactory = location.get().getExtension(HttpExecutorFactory.class);
+                Map<String, Object> httpExecutorProps = location.get().getAllConfig(true);
+                httpExecutor = httpExecutorFactory.getHttpExecutor(httpExecutorProps);
```

Comment:
Minor, could pass the location and the implementation will then decide whether to do `.getAllConfig(true);` or somthing simpler.